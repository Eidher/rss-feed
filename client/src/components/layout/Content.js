import React, { Component } from "react";
import { Route } from "react-router-dom";

import Home from "../home/Home";
import Feed from "../feed/Feed";
import Contact from "../contact/Contact";

class Content extends Component {
  scrollToTop() {
    window.scrollTo(0, 0);
    return null;
  }

  render() {
    return (
      <main>
        <div className="container h-100">
          <Route component={this.scrollToTop} />
          <Route path="/" exact component={Home} />
          <Route path="/feed" component={Feed} />
          <Route path="/contact" component={Contact} />
        </div>
      </main>
    );
  }
}

export default Content;
