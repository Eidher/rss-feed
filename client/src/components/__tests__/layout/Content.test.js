import React from "react";
import { shallow } from "enzyme";
import Content from "../../layout/Content";

describe("<Content />", () => {
  const wrapper = shallow(<Content />);

  it("should render correctly.", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should have 4 Route components", () => {
    expect(wrapper).toContainMatchingElements(4, "Route");
  });

  it("should have the right Routes and paths", () => {
    wrapper.find("Route").forEach((node, index) => {
      if (index === 0) {
        expect(node).toHaveProp("component");
      } else {
        expect(node).toHaveProp("path");
      }

      if (index === 1) {
        expect(node.prop("path")).toBe("/");
      }

      if (index === 2) {
        expect(node.prop("path")).toBe("/feed");
      }

      if (index === 3) {
        expect(node.prop("path")).toBe("/contact");
      }
    });
  });

  it("should be of type main element", () => {
    expect(wrapper.type()).toBe("main");
  });
});
