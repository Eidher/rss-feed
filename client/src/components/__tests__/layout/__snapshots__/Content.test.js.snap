// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Content /> should render correctly. 1`] = `
<main>
  <div
    className="container h-100"
  >
    <Route
      component={[Function]}
    />
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      path="/feed"
    />
    <Route
      component={[Function]}
      path="/contact"
    />
  </div>
</main>
`;
